engine/classes/ElggEntity.php

```diff
@@ -239,6 +257,15 @@ public function set($name, $value) {
 	}
 
 	/**
+	 * Get the original values of attribute(s) that have been modified since the entity was persisted.
+	 *
+	 * @return array
+	 */
+	public function getModifiedAttributes() {
```

Comment:
I think it would be less useful to get complete arrays; with only the modified value you know instantly if a change occurred and which ones changed.